### PR TITLE
Expose k8s cluster info through IK8sInfoService

### PR DESCRIPTION
**Pull request by xiaomi7732**
Address #312.

* IK8sInfoService is exposed to allow user code to access k8s cluster information when it is available;
* K8sEnvironment is made a record to ensure immutability.

I'll submit a separate PR for an example to use this feature.

### CODE REVIEW

File: src/ApplicationInsights.Kubernetes/K8sEnvironment.cs [hunk 1]
```diff
@@ -5,34 +5,39 @@
 namespace Microsoft.ApplicationInsights.Kubernetes
 {
     /// <summary>
-    /// Flatten objects for application insights or other external caller to fetch K8s properties.
+    /// An immutable flatten object for application insights or other external caller to fetch K8s properties.
```
Comment (karolz-ms) [inline src/ApplicationInsights.Kubernetes/K8sEnvironment.cs:8]:
```suggestion
    /// An immutable flat object for Application Insights or other external caller to fetch K8s properties.
```

File: src/ApplicationInsights.Kubernetes/K8sEnvironment.cs [hunk 2]
```diff
@@ -5,34 +5,39 @@
...
             V1Pod pod,
             V1ReplicaSet? replicaSet,
             V1Deployment? deployment,
-            V1Node? node
-            )
+            V1Node? node)
         {
-            _pod = pod ?? throw new System.ArgumentNullException(nameof(pod));
+            // Pod is required
+            _ = pod ?? throw new System.ArgumentNullException(nameof(pod));
```
Comment (karolz-ms) [inline src/ApplicationInsights.Kubernetes/K8sEnvironment.cs:20]: `<rant>`
We used to write in a simpler style:

```csharp
if (pod is null) {
    throw new ArgumentNullException(nameof(pod));
}
```
and I miss these days 😑 
`</rant>`

**Review by karolz-ms — APPROVED** [review]
Looks good. Don't mind my rants ☺️

**Comment by xiaomi7732** [comment]
> Looks good. Don't mind my rants ☺️

Your rants are always good rants :-D

File: src/ApplicationInsights.Kubernetes/K8sEnvironment.cs (hunk 2 above)
Comment (xiaomi7732) [reply src/ApplicationInsights.Kubernetes/K8sEnvironment.cs:20]: > `<rant>` We used to write in a simpler style:
> 
> ```cs
> if (pod is null) {
>     throw new ArgumentNullException(nameof(pod));
> }
> ```
> 
> and I miss these days 😑 `</rant>`

I thought it was generated by VSCode, it was me having pod assigned before and removed it.

Regen by code and I got exactly what you have there. I'll take this one.

**Pull request merged**